gateway-ha/src/test/java/io/trino/gateway/ha/router/TestHaRoutingManager.java

```diff
@@ -49,16 +50,16 @@ private void addMockBackends() {
       proxyBackend.setProxyTo(backend + ".trino.example.com");
       proxyBackend.setExternalUrl("trino.example.com");
       backendManager.addBackend(proxyBackend);
-      //set backend as healthyti start with
-      haRoutingManager.upateBackEndHealth(backend, true);
+      //set backend as healthy to start with
+      haRoutingManager.upateBackEndHealth(backend, BackendHealthState.HEALTHY);
     }
 
     //Keep only 1st backend as healthy, mark all the others as unhealthy
     assert (!backendManager.getAllActiveBackends().isEmpty());
 
     for (int i = 1; i < numBackends; i++) {
       backend = groupName + i;
-      haRoutingManager.upateBackEndHealth(backend, false);
+      haRoutingManager.upateBackEndHealth(backend, BackendHealthState.UNHEALTHY);
```

Comment:
nitpick: perhaps here we want to hit all enum states in the test?

for ex: 50% pending and 50% unhealthy which both represent "not yet healthy"
-> `i % 2 == 0 ? BackendHealthState.UNHEALTHY : BackendHealthState.PENDING`

(note: adding this note to all blocks as i might not be able to get back to them; feel free to resolve with or without change)